### ISSUE TITLE
withPageAuth, user always undefined

### ISSUE THREAD

**Issue by user72356** [supabase/auth-helpers#314]
Using supabase v2 and the latest version of all packages.

This example is lifted straight from the documentation. Authenticated users are properly redirected to the login page. On successful authentication, the login page redirects the user back to this page via router.replace(...) 

The issue is that the `user` variable is always `undefined`. 

```ts
// index.tsx

import { User, withPageAuth } from "@supabase/auth-helpers-nextjs";

type HomePageProps =
    {
        user: User;
    };

export default function HomePage({ user }: HomePageProps)
{
    return <div>Hello { user.email }</div>;
}

export const getServerSideProps = withPageAuth({ redirectTo: "/login" });
```

**Comment by daneburkland** [comment]
Was running into same thing, then looked at code... looks like docs are out of date  (for v.04.0). `withPageAuth` now returns an `initialSession` object, which contains `user`. See [this](https://github.com/supabase/auth-helpers/commit/0ec90f6cb3a7983e29a24ec25e6ac762005842d1#diff-168fb895e0135076b7b8b80528fb7a3a0407e89cd9afcdd92616c77687eafd39) recent diff. Could someone confirm this API change was intentional and that the docs are just out of date for v0.4.0?

**Comment by Shevly** [comment]
I too get this error, when I do getUser()

```js
{
  data: { user: null },
  error: AuthApiError: Internal server error
      at C:\Users\Riot\Documents\Project\node_modules\@supabase\gotrue-js\dist\main\lib\fetch.js:41:20
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5) {
    __isAuthError: true,
    status: 500
  }
}
```

Also no sign-in cookies are set in the browser.

**Comment by user72356** [comment]
> Was running into same thing, then looked at code... looks like docs are out of date (for v.04.0). `withPageAuth` now returns an `initialSession` object, which contains `user`. See [this](https://github.com/supabase/auth-helpers/commit/0ec90f6cb3a7983e29a24ec25e6ac762005842d1#diff-168fb895e0135076b7b8b80528fb7a3a0407e89cd9afcdd92616c77687eafd39) recent diff. Could someone confirm this API change was intentional and that the docs are just out of date for v0.4.0?

Can you provide an example of code that uses this initialSession object? I can't seem to figure out how it's being used and the types don't appear to show this object anywhere.

**Comment by bobmolgui** [comment]
Hi, you also can do **_import { useSessionContext } from '@supabase/auth-helpers-react';_** and use this in your component then as following: 

```const { error, session } = useSessionContext(); const user = session?.user;```

**Comment by user72356** [comment]
> Hi, you also can do **_import { useSessionContext } from '@supabase/auth-helpers-react';_** and use this in your component then as following:
> 
> `const { error, session } = useSessionContext(); const user = session?.user;`

By this point you might as well use the useUser() hook.

I thought the advantage of having the user or the session passed to the page as part of the page props, was to avoid this extra call. But I'm not sure how this stuff is implemented and whether there is a cost difference. I'm probably not going to bother and do a useUser(), unless someone can demonstrate an advantage to receiving it through the page props.

**Comment by nickbrinser** [comment]
> I too get this error, when I do getUser()
> 
> ```js
> {
>   data: { user: null },
>   error: AuthApiError: Internal server error
>       at C:\Users\Riot\Documents\Project\node_modules\@supabase\gotrue-js\dist\main\lib\fetch.js:41:20
>       at runMicrotasks (<anonymous>)
>       at processTicksAndRejections (node:internal/process/task_queues:96:5) {
>     __isAuthError: true,
>     status: 500
>   }
> }
> ```
> 
> Also no sign-in cookies are set in the browser.

This is the issue I'm having, and most likely the underlying problem. I *always* get an `AuthApiError` with `getUser()`, even when I also get a user back in the response. This is making `v2` pretty unusable at this point.

**Comment by daneburkland** [comment]
@Shevly @nickbrinser it appears you're running into a separate issue. This thread is about a breaking API change (not reflected in the docs) to `withPageAuth` in `v0.4.0`. See my linked diff above.

@thorwebdev can you please confirm if this new API (`withPageAuth` passing `initialSession` into props instead of `user`) will be stable going forward? Many thanks

**Comment by thorwebdev** [comment]
Thank you for reporting this, and very, sorry this was an unintended regression and has been fixed in v4.0.1: https://github.com/supabase/auth-helpers/blob/main/packages/nextjs/CHANGELOG.md#041
